bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/discovery/SunSynkAccountDiscoveryService.java

```diff
@@ -0,0 +1,104 @@
+import org.openhab.binding.sunsynk.internal.handler.SunSynkHandlerFactory;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunSynkAccountDiscoveryService} is responsible for starting the discovery procedure
+ * that connects to SunSynk Web and imports all registered inverters.
+ * 
+ * @author Lee Charlton - Initial contribution
+ */
+
+public class SunSynkAccountDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(SunSynkAccountDiscoveryService.class);
+
+    private static final int TIMEOUT = 15;
+
+    private final SunSynkAccountHandler handler;
+    private final ThingUID bridgeUID;
+
+    private ScheduledFuture<?> scanTask;
+
+    public SunSynkAccountDiscoveryService(SunSynkAccountHandler handler) {
+        super(SunSynkHandlerFactory.DISCOVERABLE_THING_TYPE_UIDS, TIMEOUT);
+        this.handler = handler;
+        this.bridgeUID = handler.getThing().getUID();
+    }
+
+    private void findInverters() {
+        List<Inverter> inverters = handler.getInvertersFromSunSynk();
+        for (Inverter inverter : inverters) {
+            addThing(inverter);
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        findInverters();
+    }
+
+    @Override
+    protected void startScan() {
+        if (this.scanTask != null) {
+            scanTask.cancel(true);
+        }
+        this.scanTask = scheduler.schedule(this::findInverters, 0, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void stopScan() {
+        super.stopScan();
+
+        if (this.scanTask != null) {
+            this.scanTask.cancel(true);
+            this.scanTask = null;
+        }
+    }
+
+    private void addThing(Inverter inverter) {
+        if (inverter == null || !inverter.discoveryInformationPresent()) {
+            return;
+        }
+
+        logger.debug("addThing(): Adding new SunSynk Inverter unit ({}) to the inbox", inverter.getAlias());
+
+        Map<String, Object> properties = new HashMap<>();
+        ThingUID thingUID = new ThingUID(SunSynkBindingConstants.THING_TYPE_INVERTER, bridgeUID, inverter.getUID());
+        properties.put(SunSynkBindingConstants.CONFIG_SECRET, inverter.getToken());
+        properties.put(SunSynkBindingConstants.CONFIG_GATE_SERIAL, inverter.getGateSerialNo());
+        properties.put(SunSynkBindingConstants.CONFIG_SERIAL, inverter.getSerialNo());
+        properties.put(Thing.PROPERTY_MODEL_ID, inverter.getID());
+        properties.put(SunSynkBindingConstants.PROPERTY_NAME, inverter.getAlias());
+
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID).withLabel(inverter.getAlias()).withBridge(bridgeUID)
+                .withProperties(properties).build());
```

Comment:
Can you add the representation property?

Reply:
Done